tests/test_pytorch/test_dynamic_shapes.py

```diff
@@ -69,82 +55,50 @@ def run_training(tmpdir, hpus, model, data_module):
         fast_dev_run=5,
     )
     trainer.fit(model, data_module())
-    return get_metric_compiles(tmpdir)
-
-
-def get_metric_compiles(dirpath):
-    """Reads metric files and returns recompiles for each trainer rank."""
-    json_files = [file for file in os.listdir(dirpath) if ".json" in file]
-    compiles = {}
-    assert json_files is not []
-    for json_file in json_files:
-        max_compiles = 0
-        with open(os.path.join(dirpath, json_file)) as file:
-            json_content = file.read()
-            # Check if the last character is ']' to determine if json is complete
-            if json_content.strip()[-1] != "]":
-                json_content += "]"
-            data = json.loads(json_content)
-            for entry in data:
-                if entry.get("metric_name") == "graph_compilation":
-                    statistics = entry.get("statistics")
-                    if statistics:
-                        total_number = statistics.get("TotalNumber")
-                        if total_number is not None and total_number > max_compiles:
-                            max_compiles = total_number
-        os.remove(os.path.join(dirpath, json_file))
-        compiles[json_file] = max_compiles
-    return compiles
-
-
-def test_dynamic_shapes_metric_file_dump(tmpdir, hpus):
-    """Tests metric file is generated."""
-    compiles = run_training(tmpdir, hpus=hpus, model=DynamicOpsBoringModel, data_module=BoringDataModule)
-    assert compiles is not {}
-
-
-def test_dynamic_shape_recompilations_recipe_caching(tmpdir, hpus):
-    """Tests number of recompilations between cached and non-cached runs."""
-    base_path = f"{tmpdir}/base"
-    compiled_path = f"{tmpdir}/compiled"
-    if not os.path.exists(base_path):
-        os.mkdir(base_path)
-    if not os.path.exists(compiled_path):
-        os.mkdir(compiled_path)
+    return gc_metric.stats()
+
 
-    os.environ["PT_HPU_METRICS_FILE"] = os.path.join(base_path, "metrics.json")
-    os.environ["PT_HPU_METRICS_DUMP_TRIGGERS"] = "process_exit,metric_change"
-    default_compiles = run_training(base_path, hpus=hpus, model=DynamicOpsBoringModel, data_module=BoringDataModule)
+def test_dynamic_shapes_recompilations_recipe_caching(tmpdir, hpus, monkeypatch):
+    """Tests number of recompilations between cached and non-cached runs."""
+    default_compiles = run_training(tmpdir, hpus=hpus, model=DynamicOpsBoringModel, data_module=BoringDataModule)
 
-    os.environ["PT_HPU_METRICS_FILE"] = os.path.join(compiled_path, "metrics.json")
```

Comment:
doc update needed ?

Reply:
no, this was only needed for getting metric dumps for recompilations, not directly related to dynamic shapes. the recompiles are now obtained using metric APIs. They have been linked in docs.